ts/src/xt.ts

```diff
@@ -0,0 +1,4066 @@
+            const currencyId = this.safeString2 (balance, 'currency', 'coin');
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            const free = this.safeString2 (balance, 'availableAmount', 'availableBalance');
+            let used = this.safeString (balance, 'frozenAmount');
+            const total = this.safeString2 (balance, 'totalAmount', 'walletBalance');
+            if (used === undefined) {
+                const crossedAndIsolatedMargin = Precise.stringAdd (this.safeString (balance, 'crossedMargin'), this.safeString (balance, 'isolatedMargin'));
+                used = Precise.stringAdd (this.safeString (balance, 'openOrderMarginFrozen'), crossedAndIsolatedMargin);
+            }
+            account['free'] = free;
+            account['used'] = used;
+            account['total'] = total;
+            result[code] = account;
+        }
+        return this.safeBalance (result);
+    }
+
+    async createOrder (symbol: string, type, side, amount, price = undefined, params = {}) {
+        /**
+         * @method
+         * @name xt#createOrder
+         * @description create a trade order
+         * @see https://doc.xt.com/#orderorderPost
+         * @see https://doc.xt.com/#futures_ordercreate
+         * @param {string} symbol unified symbol of the market to create an order in
+         * @param {string} type 'market' or 'limit'
+         * @param {string} side 'buy' or 'sell'
+         * @param {float} amount how much you want to trade in units of the base currency
+         * @param {float|undefined} price the price to fulfill the order, in units of the quote currency, can be ignored in market orders
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @param {string|undefined} params.timeInForce 'GTC', 'IOC', 'FOK' or 'GTX'
+         * @returns {object} an [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        let timeInForce = this.safeStringUpper (params, 'timeInForce', 'GTC');
+        let response = undefined;
+        if (market['spot']) {
+            request['side'] = side.toUpperCase ();
+            request['type'] = type.toUpperCase ();
+            let marginMode = undefined;
+            [ marginMode, params ] = this.handleMarginModeAndParams ('createOrder', params);
+            const marginOrSpotRequest = (marginMode !== undefined) ? 'LEVER' : 'SPOT';
+            request['bizType'] = marginOrSpotRequest;
+            if (type === 'market') {
+                timeInForce = this.safeStringUpper (params, 'timeInForce', 'FOK');
+                if (side === 'buy') {
+                    const createMarketBuyOrderRequiresPrice = this.safeValue (this.options, 'createMarketBuyOrderRequiresPrice', true);
```

Comment:
for the sake of clarity, we should add `createMarketBuyOrderRequiresPrice` to the options

Reply:
my bad then, I missed it